platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/Problem.java

```diff
@@ -21,15 +21,28 @@
 import java.util.Map;
 
 /**
- * Describes contextual information for a {@link ProblemDefinition}.
+ * Interface for describing structured information about a problem.
  */
-public interface ProblemContext {
+public interface Problem {
+
+    /**
+     * Returns the problem definition, i.e. the data that is independent of the report context.
+     */
+    ProblemDefinition getDefinition();
+
+    @Nullable
+    String getContextualLabel();
+
+    /**
+     * Returns solutions and advice that contain context-sensitive data, e.g. the message contains references to variables, locations, etc.
+     */
+    List<String> getSolutions();
```

Comment:
🤔 So you decided that the solutions should be for the individual problem, and not for the problem id?

Reply:
Yes, this was an outcome of a discussion with Adam.

Reply:
Even though some solution strings might be context-independent, conceptually, there's no way to act upon a problem definition, only on a particular instance.